addons/l10n_vn/migrations/16.0.2.0.2/post-migration.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, SUPERUSER_ID
+from odoo.osv import expression
+
+FIXED_TYPE_CODES_MAP = {
+    'expense_direct_cost': ('6271', '6272', '6273', '6274', '6277', '6278'),
+}
+
+
+def _fix_accounts_type(env):
+    vn_companies = env['res.company'].with_context(active_test=False).search([('chart_template_id', '=', env.ref('l10n_vn.vn_template').id)])
+    for correct_account_type, account_codes in FIXED_TYPE_CODES_MAP.items():
+        domain = expression.AND([
+            [('company_id', 'in', vn_companies.ids), ('account_type', '!=', correct_account_type)],
+            expression.OR([
+                [('code', 'like', f'{code}%')] for code in account_codes
+            ])
+        ])
+        accounts = env['account.account'].search(domain)
```

Comment:
I prefer it this way. By default the domain expression inside `search` is already `AND`.
Also, we usually use `=like` instead of `like` for matching account codes
```suggestion
        accounts = env['account.account'].search([
            ('company_id', 'in', vn_companies.ids),
            ('account_type', '!=', correct_account_type),
            expression.OR([
                [('code', '=like', f'{code}%')]
                for code in account_codes
            ]),
        ])